tools/backup_analysis.py

```diff
@@ -245,69 +246,93 @@ def compare_file(olderdir, newerdir, opts, filename, cmp_size):
-    chg = round(abs(chg_blocks / total_gran_blocks * 100))
-    print(f'{filename}: {total_gran_blocks} total granularity sized blocks')
-    print(f'{filename}: {chg_blocks} changed granularity blocks ({chg}%)')
-    chg = round(abs(total_bytes_diff / min_size * 100))
-    print(f'{filename}: {total_bytes_diff} of {min_size} common bytes differ ({chg}%)')
+    chg_block_pct = round(abs(chg_blocks / total_gran_blocks * 100))
+    chg_byte_pct = round(abs(total_bytes_diff / min_size * 100))
+    pct20_blocks = 0
+    pct80_blocks = 0
     if chg_blocks != 0:
         ts.files_changed += 1
         pct20_blocks = round(abs(pct20_count / chg_blocks * 100))
         pct80_blocks = round(abs(pct80_count / chg_blocks * 100))
-        print(f'{filename}: smallest 20%: {pct20_count} of {chg_blocks} changed blocks ({pct20_blocks}%) differ by {pct20} bytes or less of {opts.granularity}')
-        print(f'{filename}: largest 80%: {pct80_count} of {chg_blocks} changed blocks ({pct80_blocks}%) differ by {pct80} bytes or more of {opts.granularity}')
-    print("")
+    # Print the time even if no changes because we may want to know how long it took to not
+    # see any changes.
+    if not opts.terse:
+        print(f'{filename}: time: started {start} completed {end}')
+        if total_bytes_diff == 0:
+            # If the file is unchanged return now.
+            print(f'{filename}: is unchanged')
+            return
+
+        # Otherwise print out the change information.
+        if change_diff != 0:
+            print(f'{filename}: size: {f1_size} {f2_size} {change} by {change_diff} bytes')
+        else:
+            print(f'{filename}: size: {f1_size} {f2_size} {change}')
+        print(f'{filename}: {total_bytes_diff} of {min_size} overlapping bytes differ ({chg_byte_pct}%)')
+        print(f'{filename}: {chg_blocks} of {total_gran_blocks} granularity blocks changed ({chg_block_pct}%)')
+        if chg_blocks != 0:
+            print(f'{filename}: smallest 20%: {pct20_count} of {chg_blocks} changed blocks ({pct20_blocks}%) differ by {pct20} bytes or less of {opts.granularity}')
+            print(f'{filename}: largest 80%: {pct80_count} of {chg_blocks} changed blocks ({pct80_blocks}%) differ by {pct80} bytes or more of {opts.granularity}')
+        print("")
+    else:
+        # Print a terse summary all on one line.
+        print(
+            f'{filename}: bytes: {total_bytes_diff} of {min_size} {chg_byte_pct}%;'
+            f' blocks: {chg_blocks} of {total_gran_blocks} {chg_block_pct}%;'
+            f' {pct20_blocks}% <20%;'
+            f' {pct80_blocks}% 80>%')
 
 #
 # Print a detailed summary of all of the blocks and bytes over all of the files. We accumulated
 # the changes as we went through each of the files.
 #
 def print_summary(opts):
-    print('SUMMARY')
+    print('SUMMARY:')
     # Calculate overall totals from each of the different types first.
     total_bytes = 0
```

Comment:
Looks like `total_bytes` is no longer used?

Reply:
Fixed